dsl/dsl_cuda/tmpl8_CUDA.py

```diff
@@ -227,11 +244,11 @@
             % endfor /
 
             // Update the observable only for the last timestep
-            ##if (t == (i_step + n_step - 1)){
+            if (t == (i_step + n_step - 1)){
                 % for i, expo in enumerate(expolist):
                 tavg(i_node + ${i} * n_node) = ${expo};
                 % endfor /
-            ##}
+            }
 
             // sync across warps executing nodes for single sim, before going on to next time step
             __syncthreads();
```

Comment:
I recall this being necessary for some optimisations where multiple threads were used inside a single simulation, but this is not the case here, so do we need `__syncthreads()`? Sorry if I'm not remembering something here.